### ISSUE TITLE
Bun v1.0.31 is taking too long to install packages in Github Actions

### ISSUE THREAD

**Issue by rhuanbarreto** [oven-sh/bun#9430]
### What version of Bun is running?

v1.0.31

### What platform is your computer?

Github Actions

### What steps can reproduce the bug?

Not clear yet about how to make a correct reproduction. But my CI runs in Github actions are taking a lot of time and are timeouting during `bun install`. My bunfig file:

```toml
[install]
exact = true

[install.lockfile]
save = false
```

Another important info is that I'm using the workspaces feature in package.json:

```
"workspaces": [
  "packages/*"
],
```

So because of this monorepo arrangement, I'm not using lock files to avoid too many branch refreshes due to conflicts in bun.lockb

### What is the expected behavior?

To have the normal blazing fast bun install times.

### What do you see instead?

Timeouts of 20 minutes in my Github Actions workflows.

### Additional information

I'm using [moonrepo](https://moonrepo.dev/docs) for orchestrating the tasks in the monorepo.

Not sure if this is a clear regression but rolling back to v1.0.30 solves the issue in Github Actions.

Also, using v1.0.31 locally in a Mac (Darwin 23.3.0 arm64 arm) doesn't raise the issue locally.

I'm sending this bug report because others may have the same issue.

**Comment by fdarian** [comment]
+1 https://github.com/farreldarian/repro-bun-install/actions

![CleanShot 2024-03-15 at 16 59 21@2x](https://github.com/oven-sh/bun/assets/62016900/09e8157a-cc12-4948-a9db-084009a845d8)

**Comment by Jarred-Sumner** [comment]
Thanks, we will ship Bun v1.0.32 in about 16 hours to address this

**Comment by shiny** [comment]
I'm stuck. how solve it, instead of waiting for 16 hours?

**Comment by fdarian** [comment]
> I'm stuck. how solve it, instead of waiting for 16 hours?

Use 1.0.30

**Comment by Jarred-Sumner** [comment]
I believe this is fixed in the canary build of Bun. But there's still an outstanding shell issue we need to fix before we can push up the release.

**Comment by jamaluddinrumi** [comment]
> > I'm stuck. how solve it, instead of waiting for 16 hours?
> 
> Use 1.0.30

thank's. i'm just experienced this. and downgrade to v1.0.30 make it works again.

**Comment by shadoworion** [comment]
It also happens in AWS CodeBuild.

It stucks after: 
```
[Container] 2024/03/17 09:14:29.633376 Running command bun install
--
40 | bun install v1.0.31 (e2567512)
41 | Resolving dependencies
42 | Resolved, downloaded and extracted [2420]
```

**Comment by Jarred-Sumner** [comment]
This is fixed in Bun v1.0.32, which will release a little later today

In about an hour, if you want you can try the canary release by setting `version` in the github action to `canary`

Thank you @farreldarian for the reproduction - https://github.com/Jarred-Sumner/bun-install-bug-repro-9430/actions/runs/8316085891/job/22755142530
